### PR TITLE
Sigint 656

### PR DESCRIPTION
**Pull request by lsynopsys**

### CODE REVIEW

File: src/synopsys-action/tools-parameter.ts [hunk 1]
```diff
@@ -218,13 +218,14 @@ export class SynopsysToolsParameter {
     return command
   }
 
-  private setGithubData(): GithubData | undefined {
+  private setGithubData(isCommentFlow: boolean): GithubData | undefined {
```
Comment (sig-tithi) [inline src/synopsys-action/tools-parameter.ts:221]: why are we taking it as method param, can't we check inside as input. ?

File: src/synopsys-action/tools-parameter.ts [hunk 2]
```diff
@@ -243,14 +244,20 @@ export class SynopsysToolsParameter {
...
           branch: {
             name: githubBranchName
           }
         }
       }
+
+      if (githubPrNumber != null) {
+        githubData.repository.pull.number = Number(githubPrNumber)
+        if (isCommentFlow && isNaN(githubData.repository.pull.number)) {
+          throw new Error('Coverity/Blackduck automation PR comment can be run only by raising PR/MR')
```
Comment (sig-tithi) [inline src/synopsys-action/tools-parameter.ts:257]: Below code should be outside and before if(githubPrNumber!=null).
and Also we need to check for null.
if (isCommentFlow && isNaN(githubData.repository.pull.number)) {
          throw new Error('Coverity/Blackduck automation PR comment can be run only by raising PR/MR')
        }

File: src/synopsys-action/tools-parameter.ts (hunk 1 above)
Comment (lsynopsys) [reply src/synopsys-action/tools-parameter.ts:221]: if we use inputs then again parsing and check for both coverity and blackduck

File: src/synopsys-action/tools-parameter.ts (hunk 2 above)
Comment (lsynopsys) [reply src/synopsys-action/tools-parameter.ts]: done

File: src/synopsys-action/tools-parameter.ts (hunk 1 above)
Comment (lsynopsys) [reply src/synopsys-action/tools-parameter.ts]: updated

File: src/synopsys-action/tools-parameter.ts [hunk 3]
```diff
@@ -218,40 +218,51 @@ export class SynopsysToolsParameter {
...
+    // if there is manual run without raising pr then GITHUB_REF will return refs/heads/branch_name
+    const githubPrNumber = githubRef !== undefined ? githubRef.split('/')[2].trim() : ''
     const githubRepoOwner = process.env[FIXPR_ENVIRONMENT_VARIABLES.GITHUB_REPOSITORY_OWNER]
 
     if (githubToken == null) {
       throw new Error('Missing required github token for fix pull request/automation comment')
     }
 
+    if ((parseToBoolean(inputs.BLACKDUCK_AUTOMATION_PRCOMMENT) || parseToBoolean(inputs.COVERITY_AUTOMATION_PRCOMMENT)) && isNaN(Number(githubPrNumber))) {
+      throw new Error('Coverity/Blackduck automation PR comment can be run only by raising PR/MR')
```
Comment (sig-tithi) [inline src/synopsys-action/tools-parameter.ts:237]: Check with @Madhu if this error msg is fine " Pull request missing for Coverity or Blackduck automation PR comment. " We can modify accordingly.

**Review by sig-tithi — APPROVED** [review]
Other than error message, things looks fine.

File: src/synopsys-action/tools-parameter.ts (hunk 3 above)
Comment (lsynopsys) [reply src/synopsys-action/tools-parameter.ts]: going with "Coverity/Blackduck automation PR comment can only be triggered on a pull request."

**Pull request merged**